single_kernel_mongo/core/secrets.py

```diff
@@ -21,14 +21,16 @@
 logger = logging.getLogger(__name__)
 
 
-def generate_secret_label(app_name: str, scope: Scope) -> str:
+def generate_secret_label(app_name: str, scope: Scope, relation: str | None = None) -> str:
     """Generate unique group_mappings for secrets within a relation context.
 
     Defined as a standalone function, as the choice on secret labels definition belongs to the
     Application Logic. To be kept separate from classes below, which are simply to provide a
     (smart) abstraction layer above Juju Secrets.
     """
     members = [app_name, scope.value]
+    if relation:
+        members = [relation, app_name, scope.value]
```

Comment:
nit
```suggestion
    members = [app_name, scope.value]
    if relation:
        members.append(relation)
```

reason: easier to see the diff

Reply:
This is not the same behavior as it would put the relation at the end, which differs from what is done everywhere else: `<relation>.<app>.<scope>`.
I agree that this is not the best diff but I'm trying to stick with the standards :)

Reply:
ah - thanks for the explaination. No worries then!